cmd/setup.go

```diff
@@ -44,7 +42,7 @@ func getGRPCMiddleware(cfg *config.Shield, logger log.Logger) grpc.ServerOption
 	return grpc.UnaryInterceptor(
 		grpcMiddleware.ChainUnaryServer(
 			grpc_interceptors.EnrichCtxWithIdentity(cfg.App.IdentityProxyHeader),
-			grpczap.UnaryServerInterceptor(zap.NewExample()),
+			grpczap.UnaryServerInterceptor(logger.(*log.Zap).GetInternalZapLogger().Desugar()),
```

Comment:
@bsushmith we can make it more defensive by always checking the casting and have a fallback scenario for this

Reply:
Then, in that case, would the default fallback be `zap.NewExample()` itself? 🤔